ddtrace/tracer/remote_config_test.go

```diff
@@ -121,6 +123,53 @@ func TestOnRemoteConfigUpdate(t *testing.T) {
 		telemetryClient.AssertCalled(t, "ConfigChange", []telemetry.Configuration{{Name: "trace_header_tags", Value: "", Origin: ""}})
 	})
 
+	t.Run("RC tracing_enabled = false is applied", func(t *testing.T) {
+		telemetryClient := new(telemetrytest.MockClient)
+		defer telemetry.MockGlobalClient(telemetryClient)()
+		t.Setenv("DD_APPSEC_ENABLED", "true")
+
+		Start(WithService("my-service"), WithEnv("my-env"))
+		defer Stop()
+
+		appsec.Start()
+		defer appsec.Stop()
+
+		input := remoteconfig.ProductUpdate{
+			"path": []byte(`{"lib_config": {"tracing_enabled": false}, "service_target": {"service": "my-service", "env": "my-env"}}`),
+		}
+
+		tr, ok := internal.GetGlobalTracer().(*tracer)
+		require.Equal(t, true, ok)
+		applyStatus := tr.onRemoteConfigUpdate(input)
+		require.Equal(t, state.ApplyStateAcknowledged, applyStatus["path"].State)
+		require.Equal(t, false, tr.config.enabled.current)
+		require.Equal(t, true, appsec.Enabled())
+		headers := TextMapCarrier{
+			traceparentHeader:      "00-12345678901234567890123456789012-1234567890123456-01",
+			tracestateHeader:       "dd=s:2;o:rum;t.usr.id:baz64~~",
+			"ot-baggage-something": "someVal",
+		}
+		sctx, err := tr.Extract(headers)
+		require.NoError(t, err)
+		require.Equal(t, internal.NoopSpanContext{}, sctx)
+		err = tr.Inject(internal.NoopSpanContext{}, TextMapCarrier{})
+		require.NoError(t, err)
+		require.Equal(t, internal.NoopSpan{}, tr.StartSpan("noop"))
+
+		// all subsequent spans are of type internal.NoopSpan
+		// no further remoteConfig changes are applied
+		s := StartSpan("web.request").(internal.NoopSpan)
+		s.Finish()
+
+		// turning tracing back on is not allowed
+		input = remoteconfig.ProductUpdate{"path": nil}
```

Comment:
This test looks wrong to me. Should it be something like this:
```
		input = remoteconfig.ProductUpdate{
			"path": []byte(`{"lib_config": {"tracing_enabled": true}, "service_target": {"service": "my-service", "env": "my-env"}}`),
		}
```
which explicitly attempts to turn `tracing_enabled` back to `true` and verify that the config doesn't change? In fact, when I run the test with the above remote config input, there is an error at line 170 where it expects 1 telemetry call but gets 2 (which is possibly the bug I commented about in the other file))

Reply:
I can add this one too, but I wanted to have nil because it's akin to 'reset' case.  It is also in line with other RC tests

Reply:
Okay, let's keep this test but also have explicitly enabling it too.
Though like I said, I think there might be a small bug that I described here https://github.com/DataDog/dd-trace-go/pull/2513/files#r1486645555 which will make the `true` case fail

Reply:
Yes, indeed. Added that test case